server/models/questions/addEditor.js

```diff
@@ -0,0 +1,33 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {String} param0.editor
+ * @param {Number} param0.questionId
+ * @return {Promise}
+ */
+
+function addEditor({ username, editor, questionId }) {
+  return new Promise((resolve, reject) => {
+    pool.query(
+      `INSERT INTO questions_editors(question_id, editor) 
+       SELECT question_id, ? FROM questions_editors WHERE editor=? AND question_id=?`,
+      [editor, username, questionId],
+      (error, results) => {
+        if (error || !results) {
+          return reject(error)
+        }
+
+        if (!results.affectedRows) {
+          return reject('Invalid question_id or unauthorized user')
```

Comment:
```suggestion
          return reject('Invalid question Id or the user do not have required permissions.')
```
The term 'unauthorized' might be confusing as not logged in.

Reply:
It is authentication that is related to users to be logged in isn't it ? Being unauthorized means a particular resource is not accessible to that user

Reply:
"user does not have required permissions" = "unauthorized"

Reply:
Still, a better error message would be- (this would be directly displayed in frontend, easy to understand for the user)
```
 return reject('Invalid question Id or you do not have required permissions.')
```